### PR TITLE
fix(anon): add regional-intelligence + trade-policy to WEB_PREMIUM_PANELS (lock-CTA invariant)

### PR DESCRIPTION
**Pull request by koala73**
## Summary

`src/config/panels.ts` and `src/app/panel-layout.ts` carry **two premium-related lists** that have to stay in sync:

| List | Lives in | Purpose |
|---|---|---|
| `apiKeyPanels` | `panels.ts` `isPanelEntitled()` | API-key holders + Pro users may mount the panel |
| `WEB_PREMIUM_PANELS` | `panel-layout.ts` | `updatePanelGating()` → `Panel.showGatedCta()` paints "Sign In to Unlock" / "Upgrade to Pro" for non-premium |

`regional-intelligence` and `trade-policy` were in (a) but missing from (b). API-key users got the panels correctly. **Anonymous / free-Clerk web users got the panel mounted, its loader fired — writing empty / loading / error UI directly into the body — while the PRO badge in the header still rendered.** The combined "PRO + visible internal loader/empty copy" looks broken.

Concrete regression: PR #3578 added a soft empty state to `RegionalIntelligenceBoard`. For anonymous users it surfaced *"Regional intelligence is being refreshed. Try selecting another region above."* instead of the lock CTA. The existing `panel-config-guardrails.test.mjs` even has a comment claiming `regional-intelligence` was gated via `WEB_PREMIUM_PANELS` — the comment was aspirational; the actual set entry never landed.

Same root cause for Trade Policy stuck on "Loading…" forever for anonymous.

## Behaviour change

| User type | Before | After |
|---|---|---|
| Anonymous | Panel mounts, loader runs, body shows empty/loading state, PRO badge in title (looks broken) | `showGatedCta(ANONYMOUS)` — "Sign In to Unlock" CTA |
| Free Clerk | Same broken state as anon | `showGatedCta(FREE_TIER)` — "Upgrade to Pro" CTA |
| API-key holder (desktop) | Unchanged — panel works | Unchanged — `unlockPanel()` |
| Pro user | Unchanged | Unchanged |

## Invariant test

New `tests/panel-config-guardrails.test.mjs` assertion: `apiKeyPanels ⊆ WEB_PREMIUM_PANELS`. Verified by stashing only the `panel-layout.ts` change and running the test — it fails with exactly the two orphan keys before the fix, passes after:

```
AssertionError: apiKeyPanels members missing from WEB_PREMIUM_PANELS:
  regional-intelligence, trade-policy
```

This blocks the same drift from recurring silently.

## Tracks

`todos/257-pending-p2-anon-broken-panels-sweep.md` — items **2** (Trade Policy) and **8** (Regional Intelligence). Item 3 (WSB Ticker retry-state for anon) is unrelated — `wsb-ticker-scanner` was already in `WEB_PREMIUM_PANELS`. Likely a stale-state-on-signout edge case for that one; investigate separately.

## Test plan

- [x] `npm run typecheck` — PASS
- [x] `npm run typecheck:api` — PASS
- [x] `npm run lint` — PASS (warnings only)
- [x] `npm run test:data` — PASS (7848/7848, +1 new invariant test)
- [x] New invariant test passes ON the fix and FAILS without it (drift detection works)
- [x] Pre-push strict gate (boundaries, rate-limit, premium-fetch, edge-bundles, version) — PASS
- [ ] Manual: load app anonymously → Regional Intelligence shows "Sign In to Unlock", not "is being refreshed"
- [ ] Manual: load app anonymously → Trade Policy shows "Sign In to Unlock", not stuck loading
- [ ] Manual: log in as Pro → both panels populate as before

### CODE REVIEW

File: tests/panel-config-guardrails.test.mjs [hunk 1]
```diff
@@ -87,6 +87,50 @@ describe('panel-config guardrails', () => {
...
+    // wrote "Regional intelligence is being refreshed" into the body
+    // because regional-intelligence was in apiKeyPanels (so isPanelEntitled
+    // mounted it) but missing from WEB_PREMIUM_PANELS (so showGatedCta
+    // never fired). See todos/257-pending-p2-anon-broken-panels-sweep.md
+    // item 8.
+    const panelsSrc = readFileSync(resolve(__dirname, '../src/config/panels.ts'), 'utf-8');
+
+    const apiKeyPanelsMatch = panelsSrc.match(/const apiKeyPanels = \[([^\]]+)\];/);
+    assert.ok(apiKeyPanelsMatch, 'apiKeyPanels array not found in panels.ts');
+    const apiKeyPanels = [...apiKeyPanelsMatch[1].matchAll(/'([^']+)'/g)].map(m => m[1]);
```
[the remark this answers is not in the document]
Comment (koala73) [reply tests/panel-config-guardrails.test.mjs]: Addressed in commit 3c3b3e38e — hoisted a shared `QUOTED` regex (`/['"]([^'"]+)['"]/g`) used for both `apiKeyPanels` and `WEB_PREMIUM_PANELS` extraction, so the two halves stay symmetric and the guard outlives any future quote-style drift even though biome currently enforces single quotes. (Note: the suggestion as posted dropped the closing `]` in the character class — corrected to `['"]([^'"]+)['"]` in the landed change.)

File: tests/panel-config-guardrails.test.mjs [hunk 2]
```diff
@@ -87,6 +87,50 @@ describe('panel-config guardrails', () => {
...
+    assert.ok(apiKeyPanels.length > 0, 'apiKeyPanels parse returned no entries');
+
+    const webPremiumMatch = panelLayoutSrc.match(/const WEB_PREMIUM_PANELS = new Set\(\[([\s\S]*?)\]\);/);
+    assert.ok(webPremiumMatch, 'WEB_PREMIUM_PANELS not found in panel-layout.ts');
+    const webPremium = new Set([...webPremiumMatch[1].matchAll(/'([^']+)'/g)].map(m => m[1]));
```
[the remark this answers is not in the document]
Comment (koala73) [reply tests/panel-config-guardrails.test.mjs]: Same fix in commit 3c3b3e38e — both extractions now use the shared `QUOTED` regex, so they stay symmetric and accept both quote styles.
